### ISSUE TITLE
put our todo transactions in a transaction

### ISSUE THREAD

**Issue by phritz** [rocicorp/replicache-old#70]
tldr: our todo data layer transactions are not actually transactional

from slack:

ok then we push the del mutation id 402 and then pull. the pull returns last mutation id of 402 but not a patch that deletes the value. this is not expected. the diffserver log shows that the client view returned last mutation id of 402 and all the todos, including the one we just deleted. so you are right, it is an isolation level thing. i went looking and turns out we are holding the aws calls wrong. in particular execute statement https://docs.aws.amazon.com/rdsdataservice/latest/APIReference/API_ExecuteStatement.html which we call to execute our BEGIN, queries, and COMMIT/ROLLBACK https://github.com/rocicorp/replicache-sample-todo/blob/6764fe56c5f99ef91c9dea727e5cbac4d8ba98b3/serve/db/db.go#L61 has the default behavior of autocommitting each statement unless we pass in a transaction id, which we don't. so our changes are not actually transactional. unit test doesnt check that transactions are isolated which is understandable because thats what databases are supposed to do :slightly_smiling_face: so anyway, solution is to call begintransaction to get a txid and pass it through into the executestatement params, and to use committransaction and rollbacktransaction to do those operations (instead of executestatement).
